julia/LibCEED.jl/src/Misc.jl

```diff
@@ -0,0 +1,119 @@
+import LinearAlgebra: det
+
+"""
+    CeedDim(dim)
+
+The singleton object of type `CeedDim{dim}`, used for dispatch to linear algebra operations
+specialized for small matrices (1, 2, or 3 dimensions).
+"""
+struct CeedDim{dim} end
+@inline CeedDim(dim) = CeedDim{Int(dim)}()
+
+"""
+    det(J, ::CeedDim{dim})
+
+Specialized determinant calculations for matrices of size 1, 2, or 3.
+"""
+@inline det(J, ::CeedDim{1}) = @inbounds J[1]
+@inline det(J, ::CeedDim{2}) = @inbounds J[1]*J[4] - J[3]*J[2]
+#! format: off
+@inline det(J, ::CeedDim{3}) = @inbounds (
+    J[1]*(J[5]*J[9] - J[6]*J[8]) -
+    J[2]*(J[4]*J[9] - J[6]*J[7]) +
+    J[3]*(J[4]*J[8] - J[5]*J[7])
+)
+#! format: on
+
+"""
+    setvoigt(J::StaticArray{Tuple{D,D},T,2})
+    setvoigt(J, ::CeedDim{dim})
+
+Given a symmetric matrix `J`, return a `SVector` that encodes `J` using the [Voigt
+convention](https://en.wikipedia.org/wiki/Voigt_notation).
```

Comment:
Thanks for adding the reference here.

Just curious. I've seen this question come up somewhere else. Have you ever found in your experience a way to insert citations from a bibtex file and have them rendered nicely in Julia's docstrings?

Reply:
I haven't found anything good for bibtex references. I think there are a few "homegrown" solutions, but nothing official. There is a proposal to implement something in Documenter.jl, so I expect that it will happen at some point.

Reply:
Good to know. Thanks!

Reply:
Just to follow up. I believe this would do it
https://github.com/ali-ramadhan/DocumenterCitations.jl

Reply:
Very cool, this could be useful. Thanks!